src/KafkaW/Consumer.cpp

```diff
@@ -127,17 +133,36 @@ bool Consumer::topicPresent(const std::string &TopicName) {
   return true;
 }
 
+/// Updates the RdKafka Metadata pointer. If broker is unavailable, keeps trying
+/// to connect every 500ms, logging messages every few seconds.
 void Consumer::updateMetadata() {
+  short LoopCounter = 0;
+  while (!metadataCall()) {
```

Comment:
Although this function is very simple, I still think that we should have one or two unit tests for it. It should be easy (I think) to write a simple test that mocks the call to `metadataCall()`.